Classes/Security/FilePermissionsAspect.php

```diff
@@ -24,18 +26,30 @@
 class FilePermissionsAspect {
 
 	/**
-	 * Post process the matcher object.
+	 * Postprocess the matcher object to respect file storages.
 	 *
 	 * @param Matcher $matcher
 	 * @param string $dataType
 	 * @return void
 	 */
-	public function addFilePermissions(Matcher $matcher, $dataType) {
+	public function addFilePermissionsForFileStorages(Matcher $matcher, $dataType) {
 		if ($dataType === 'sys_file') {
 			$this->respectStorage($matcher);
+		}
+	}
 
+	/**
+	 * Postprocess the constraints object to respect file mounts.
+	 *
+	 * @param Query $query
+	 * @param ConstraintInterface|NULL $constraints
+	 * @param string $dataType
+	 * @return void
+	 */
+	public function addFilePermissionsforFileMounts(Query &$query, &$constraints, $dataType) {
```

Comment:
I am not sure why it seems necessary to use "&" for $query and $constraints here. AFAIK objects should automatically be passed by reference in PHP. But without those "&" it will not work.